tools/sorin_parser/UserFunLexer.mll

```diff
@@ -0,0 +1,26 @@
+(* File lexer.mll *)
+{
+	open UserFunParser
+	exception Eof
+}
+rule token = parse
+    [' ' '\t']  { token lexbuf }
+  | ['\n' ]     { EOL }
+  | ['0'-'9'] ['0'-'9']* as lxm { INT(lxm) }
+  | ['A'-'Z' 'a'-'z']+ as lxm { VAR(lxm) }
+  | '+'         { PLUS }
+  | '-'			    { MINUS }
+  | '*'         { TIMES }
+  | '/'         { DIV }
+  | '%'         { MOD }
+  | '='         { EQ }
+  | '>'         { GT }
+  | '<'         { LT }
+  | '('         { LPAREN }
+  | ')'         { RPAREN }
+  | "!="        { NEQ }
+  | "<="        { LTEQ }
+  | ">="        { GTEQ }
+  | "||"        { OR }
+  | "&&"        { AND }
+  | eof         { raise Eof }
```

Comment:
Something to check: Do we parse all their files without errors? I am asking because I don't see a `NOT` token, and I am not sure if it's allowed by them or not.